### DISCUSSION TITLE
Returning the SHAP for a single feature

### DISCUSSION THREAD

**Discussion by rcwoolston** [shap/shap#3117] [Q&A] [answered]
I would like to include SHAP values with more granularly in a report and leave space for comment, is there a way to break the summary plot into n, where n is the number of features, plots? Where each plot is a single feature.

**Comment by thatlittleboy** [comment, marked as answer]
In short, not elegantly at the moment. 
Right now, it is hardcoded for each feature to take up "1" unit along the y-axis, and it's not possible currently to adjust this spacing.

Naively, you can easily plot for a single feature only multiple times:

```python
import xgboost
import shap

# train XGBoost model
X,y = shap.datasets.adult(n_points=500)
model = xgboost.XGBClassifier(n_estimators=100).fit(X, y)

# compute SHAP values
explainer = shap.Explainer(model, X)
shap_values = explainer(X)

for c in ("Age", "Relationship"):
    shap.plots.beeswarm(shap_values[:, [c]], max_display=1)
```

![image](https://github.com/shap/shap/assets/30731072/aade7f06-451e-4d3a-a6f9-fd512a593077)

You can then modify the plots however you like, e.g. remove the xlabels and colorbars, etc.

There was also a request recently #3070 to expose the `ax` parameter, so you could pre-define subplot axes of a fixed size, and plot the various beeswarm plot in those axes, so at least the plot sizes can be consistent for all your individual features.
